### PR TITLE
feat: add quantum leap discount

### PR DESCRIPTION
**Pull request by fricoben**
This PR adds a potential Quantum Leap discount but also a way to create discounts easily with offer-upsell-checkout. 

I would like the review to be really focused on my architecture concerning the replicability of making a discount campaign and also the fact that I don't break important components (like the search bar for example). 

Note: the email part is not managed for the moment

Thanks !

### CODE REVIEW

File: components/UI/searchBar.tsx [hunk 1]
```diff
@@ -155,6 +159,13 @@ const SearchBar: FunctionComponent<SearchBarProps> = ({
         message: valid + " is not a valid caracter",
         lastAccessed: lastAccessed ?? Date.now(),
       };
+    } else if (is5LettersOnly && name.length < 5) {
+      return {
+        name,
+        error: true,
+        message: "Only 5 letters domains for this discount",
```
Comment (irisdv) [inline components/UI/searchBar.tsx:166]: Currently the message is also shown on the homepage when you search a domain

File: utils/discounts/quantumLeap.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  discountMailGroupId: string;
+};
+
+export const quantumLeapDiscount: QuantumLeapDiscount = {
+  offer: {
+    duration: 81,
+    customMessage: "3 months (with Quantum Leap discount)",
+    discountId: "1",
+    price: "599178082191783",
+    desc: "To celebrate the Quantum Leap, get your domain for just $1 and unlock a world of possibilities. Customize your On-chain StarkNet username, elevate your blockchain us er experience, and access domain benefits.",
```
Comment (irisdv) [inline utils/discounts/quantumLeap.ts]: there is a space in the middle of the word `user` in `your blockchain us er experience`

File: utils/discounts/quantumLeap.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      desc: "Special Offer:",
+      catch: "Get 3 years for the price of 2",
+    },
+    image: "/quantumleap/quantumLeapAstro2.webp",
+  },
+  name: "The Quantum Leap",
+  image: "/quantumleap/quantumLeapAstro.webp",
+  expiry: 1694908799 * 1000,
+  discountMailGroupId: "98859014745490932",
+} as const;
```
Comment (irisdv) [inline utils/discounts/quantumLeap.ts]: it's already declared as a const

File: components/discount/discountCheckoutScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        <RegisterDiscount
+          domain={domain}
+          duration={duration}
+          discountId={discountId}
+          customMessage={customMessage}
+          price={price}
+          mailGroups={[
+            process.env.NEXT_PUBLIC_MAILING_LIST_GROUP ?? "",
+            mailGroupId,
+          ]} // Second group is the special group for
```
Comment (irisdv) [inline components/discount/discountCheckoutScreen.tsx]: I think there's a word missing at the end of the comment

File: utils/discounts/quantumLeap.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  upsell: Discount;
+  expiry: number;
+  image: string;
+  name: string;
+  discountMailGroupId: string;
+};
+
+export const quantumLeapDiscount: QuantumLeapDiscount = {
+  offer: {
+    duration: 81,
```
Comment (irisdv) [inline utils/discounts/quantumLeap.ts]: What's the unit of the duration ?

**Review by irisdv — CHANGES_REQUESTED** [review]
Your architecture looks good to me! The only case I see that would require more changes, would be if a discount is made on domains that have a length lower than 5 characters, it would require to update the code of the searchbar. If that could be the case one day, maybe you could update the search bar to pass the number of characters, instead of having `is5LettersOnly`.

I left a few comments in the code and I noticed that on mobile the register form with the discount is not responsive.

File: utils/discounts/quantumLeap.ts (hunk 5 above)
Comment (fricoben) [reply utils/discounts/quantumLeap.ts]: Days, would be better to say it in the comment or in the name of the variable ?

Comment (irisdv) [reply utils/discounts/quantumLeap.ts]: yes I think a comment would be good just in case someone new needs to create a new discount

**Review by irisdv — APPROVED** [review]
lgtm!

**Pull request merged**